core/iwasm/libraries/thread-mgr/thread_manager.c

```diff
@@ -1302,6 +1307,30 @@ wasm_cluster_set_exception(WASMExecEnv *exec_env, const char *exception)
     data.skip = NULL;
     data.exception = exception;
 
+#if WASM_ENABLE_DUMP_CALL_STACK != 0
+    if (has_exception) {
+        /* Save the stack trace of the crashed thread into the cluster */
+        WASMModuleInstance *module_inst =
+            (WASMModuleInstance *)get_module_inst(exec_env);
+
+#if WASM_ENABLE_INTERP != 0
+        if (module_inst->module_type == Wasm_Module_Bytecode
+            && wasm_interp_create_call_stack(exec_env)) {
+            wasm_frame_vec_clone_internal(module_inst->frames,
+                                          &cluster->exception_frames);
```

Comment:
Two threads may set exception simultaneously, and access `cluster->exception_frames` at the same time? Had better move these lines to be after `os_mutex_lock(&cluster->lock)` below?